programs/mango-v4/src/state/bank.rs

```diff
@@ -1065,14 +1064,13 @@ impl Bank {
         self.interest_curve_scaling = (self.interest_curve_scaling * adjustment).max(1.0)
     }
 
-    pub fn oracle_price(
+    pub fn oracle_price<T: KeyedAccountReader>(
         &self,
-        oracle_acc: &impl KeyedAccountReader,
+        oracle_acc_infos: &OracleAccountInfos<T>,
         staleness_slot: Option<u64>,
     ) -> Result<I80F48> {
```

Comment:
honest question: now that oracle_price and oracle_price_with_fallback have the same signature, do we even still need both? Maybe it's enough to say "the fallback is checked, if `oracle_acc_infos.fallback.is_some()`"?